server/docker_resource.py

```diff
@@ -0,0 +1,433 @@
+        for image in current.getDockerImageList():
+            data[image] = current.getCLIDict(image)
+        return data
+
+    @access.admin
+    @describeRoute(
+        Description('Remove a docker image ')
+        .notes(
+            """Must be a system administrator to call this. If the value
+                       passed is a valid JSON object.""")
+        .param('name', 'The name or a list of names  of the '
+                       'docker images to be removed', required=True)
+        .errorResponse('You are not a system administrator.', 403)
+        .errorResponse('Failed to set system setting.', 500)
+    )
+    # TODO delete REST endpoint
+    def deleteImage(self, params):
+
+        self.requireParams(('name',), params)
+        name = params['name']
+        name = json.loads(name)
+
+        if isinstance(name, list):
+            for img in name:
+
+                if not isinstance(img, six.string_types):
+                    raise RestException('%s was not a valid string.' % img)
+            else:
+                self._deleteImage(name)
+        elif isinstance(name, six.string_types):
+
+            self._deleteImage([name])
+
+        else:
+
+            raise RestException('name was not a valid JSON list or string.')
+
+    def _deleteImage(self, names):
+        """
+        Removes the docker images and there respective clis from the settings
+        :param name: The name of the docker image (user/rep:tag)
+
+        """
+        currentSettings = DockerCache(getDockerImageSettings())
+        for name in names:
+
+            if not currentSettings.deleteImage(name):
+                raise RestException('%s does not exist' % name)
+        # need to remove each cli attribute
+        # need to remove the clli list attribute
+        # need to remove the rest route
+        print "after delete", currentSettings.raw()
+        ModelImporter.model('setting').set(PluginSettings.DOCKER_IMAGES,
+                                           currentSettings.raw())
+
+    @access.admin
+    @describeRoute(
+        Description('Add a to the list of images to be loaded ').notes(
+            """Must be a system administrator to call this. If the value
+                   passed is a valid JSON object, it will be parsed and stored
+                   as an object.""").param(
+            'name', 'The name or a list of names  of the '
+                    'docker images to be loaded ', required=True)
+            .errorResponse('You are not a system administrator.', 403)
+            .errorResponse('Failed to set system setting.', 500)
+    )
+    # TODO check the local cache and cloud for different images of same name
+    # TODO how to handle newer images(take the latest or require a confirmation)
+    # TODO use image id to confirm equivalence need v2 manifest schema on cloud
+    # TODO how to handle duplicate clis
+    # TODO create the new REST endpoints
+    def appendImage(self, params):
```

Comment:
@cdeepakroy 
I am considering setImages just because calling this REST endpoint removes previous images and attempts to set only the ones passed in as a param as the new available docker images